main.py

```diff
@@ -212,6 +293,650 @@ def get_job_details(job_id: int) -> str:
+            min_idle = pool.get("min_idle_instances", 0)
+            max_capacity = pool.get("max_capacity", 0)
+            state = pool.get("state", "N/A")
+            
+            table += f"| {pool_id} | {pool_name} | {instance_type} | {min_idle} | {max_capacity} | {state} |\n"
+        
+        return table
+    except Exception as e:
+        return f"Error listing instance pools: {str(e)}"
+
+@mcp.tool()
+def create_instance_pool(
+    name: str, 
+    node_type_id: str, 
+    min_idle_instances: int = 0, 
+    max_capacity: int = 10,
+    idle_instance_autotermination_minutes: int = 60,
+    preload_spark_versions: List[str] = None
+) -> str:
+    """Create a new instance pool
+    
+    Args:
+        name: The name of the instance pool
+        node_type_id: The node type for the instances in the pool (e.g., "Standard_DS3_v2")
+        min_idle_instances: Minimum number of idle instances to keep in the pool
+        max_capacity: Maximum number of instances the pool can contain
+        idle_instance_autotermination_minutes: Number of minutes that idle instances are maintained
+        preload_spark_versions: Spark versions to preload on the instances (optional)
+    """
+    try:
+        # Build the request data
+        data = {
+            "instance_pool_name": name,
+            "node_type_id": node_type_id,
+            "min_idle_instances": min_idle_instances,
+            "max_capacity": max_capacity,
+            "idle_instance_autotermination_minutes": idle_instance_autotermination_minutes
+        }
+        
+        # Add preloaded Spark versions if specified
+        if preload_spark_versions:
+            data["preloaded_spark_versions"] = preload_spark_versions
+        
+        # Make the API request
+        response = databricks_api_request("instance-pools/create", method="POST", data=data)
+        
+        # The response contains the instance_pool_id of the newly created pool
+        pool_id = response.get("instance_pool_id")
+        
+        if pool_id:
+            return f"Instance pool created successfully with ID: {pool_id}"
+        else:
+            return "Instance pool creation failed. No pool ID returned."
+    except Exception as e:
+        return f"Error creating instance pool: {str(e)}"
+
+# DBFS API tools
+@mcp.tool()
+def list_dbfs_files(path: str = "/") -> str:
+    """List files in a DBFS directory"""
+    try:
+        if not all([DATABRICKS_HOST, DATABRICKS_TOKEN]):
+            raise ValueError("Missing required Databricks API credentials in .env file")
+        
+        # Direct API call to match the exact curl command that works
+        import subprocess
+        import json
+        
+        # Execute the curl command that we know works
+        cmd = [
+            "curl", "-s",
+            "-H", f"Authorization: Bearer {DATABRICKS_TOKEN}",
+            "-H", "Content-Type: application/json",
+            f"https://{DATABRICKS_HOST}/api/2.0/dbfs/list?path={path}"
+        ]
```

Comment:
I think here we can use the request function as in the other calls. Not sure why to do it using subprocess and cmd. 

[Databricks API call](https://docs.databricks.com/api/workspace/dbfs/list)